fhttp/http_client.go

```diff
@@ -380,6 +380,20 @@ func newHTTPRequest(o *HTTPOptions) (*http.Request, error) {
 	}
 	//nolint:noctx // we pass context later in Run()/Fetch()
 	req, err := http.NewRequest(method, o.URL, body)
+	if err == nil { //nolint:nestif
+		// Additional validation for the URL so we abort early on fatal errors even for the std client.
+		// fixes #784
+		if req.URL == nil || req.URL.Host == "" {
+			err = fmt.Errorf("invalid url '%s'", o.URL)
```

Comment:
It took me (too) long to realise that you are setting `err` here for line 383 if I'm honest. Same goes for the logic at line 394. 

I do understand you want to reuse the contex of the logger... but then maybe make a small func as in "return err with context" and then do something like

```
return nil, returnErrorWithContext("invalid url")
```

where `returnErrorWithContext`  func is the contents of:

```
		log.S(log.Error, "Unable to make request",
			log.Attr("method", method), log.Attr("url", o.URL), log.Attr("err", err),
			log.Attr("thread", o.ID), log.Attr("run", o.UniqueID))
```

Reply:
yeah it was to avoid repeating / share the log.S indeed. but you're right it'd be a bit clearer/cleaner with a function, specially if used in more places. there is a small downside though which is the line number in the log will be the one of the function (though maybe that's also something I could try to change, expose the depth)

Reply:
Will address (maybe ;-) in #787